core/types/calls.cc

```diff
@@ -800,7 +825,7 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, const DispatchArgs &arg
             auto kwSplatType = Types::approximate(gs, kwSplatArg->type, *constr);
 
             if (hasKwargs) {
-                if (auto *hash = cast_type<ShapeType>(kwSplatType)) {
+                if (auto *hash = isKwargsHash(gs, kwSplatType)) {
```

Comment:
Fallback behavior when this is null is that we pass the kwargsplatted hash as-is as a posarg, right?

How would the following example be handled?

https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7Bparams(h%3A%20T%3A%3AHash%5BT.untyped%2C%20Integer%5D).void%7D%0Adef%20foo(h)%0Aend%0A%0Ah%20%3D%20%7B42%20%3D%3E%2043%7D%0Afoo(**h%2C%20x%3A%20222)%0A

Reply:
In this case we'll behave like ruby, and treat this as a single hash passed as positional argument.